source/common/upstream/load_balancer_impl.cc

```diff
@@ -593,14 +593,32 @@ HostConstSharedPtr EdfLoadBalancerBase::chooseHostOnce(LoadBalancerContext* cont
 
 HostConstSharedPtr LeastRequestLoadBalancer::unweightedHostPick(const HostVector& hosts_to_use,
                                                                 const HostsSource&) {
-  // This is just basic unweighted P2C.
-  const HostSharedPtr host1 = hosts_to_use[random_.random() % hosts_to_use.size()];
-  const HostSharedPtr host2 = hosts_to_use[random_.random() % hosts_to_use.size()];
-  if (host1->stats().rq_active_.value() < host2->stats().rq_active_.value()) {
-    return host1;
-  } else {
-    return host2;
+  // A user is not allowed to specify less than two choices.
+  uint32_t choice_count = 2;
```

Comment:
I would just go ahead and store a constant choice_count in the constructor directly, and not store the config object which we don't need. Avoids some extra instructions on the hot path.